util-taglib/src/com/liferay/taglib/ui/InputFieldTag.java

```diff
@@ -98,8 +101,34 @@ public boolean isIgnoreRequestValue() {
 		return _ignoreRequestValue;
 	}
 
-	public void setAutoComplete(boolean autoComplete) {
-		_autoComplete = autoComplete;
+	public void setAutoComplete(String autoComplete) {
+		autoComplete = StringUtil.toLowerCase(autoComplete.trim());
+
+		String[] autoCompleteParts = StringUtil.split(
+			autoComplete, CharPool.SPACE);
+
+		boolean validAutoComplete = false;
+
+		if (autoCompleteParts.length == 1) {
+			validAutoComplete = _validateAutoCompleteValue(
+				autoCompleteParts[0]);
+		}
+		else if (autoCompleteParts.length == 2) {
+			validAutoComplete =
+				_validateAutoCompleteValue(autoCompleteParts[1]) &&
+				(_validateAutoCompleteContactInformation(
+					autoCompleteParts[0]) ||
+				 _validateSectionAutoComplete(autoCompleteParts[0]));
+		}
+		else if (autoCompleteParts.length == 3) {
```

Comment:
If you decide to go with full spec, I think this is missing something... If I read it correctly this would look like:

`("on"|"off") | [section-*] [shipping|billing] (name|prefix|...)|[home|work|mobile|fax|pager](tel|email|...)`

That is, when we don't pass "on" or "off", we can have up to 4 tokens which you can see represented in the [processing model](https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#autofill-processing-model) section